components/Send/Send.tsx

```diff
@@ -99,66 +120,277 @@ const Send: React.FunctionComponent<SendProps> = ({
+      toAddr.amountCurrency = amountCurrency.substring(0, 15);
+      if (isNaN(Utils.parseStringLocaleToNumberFloat(toAddr.amountCurrency))) {
+        toAddr.amount = '';
+      } else if (toAddr.amountCurrency && zecPrice && zecPrice.zecPrice > 0) {
+        toAddr.amount = Utils.parseNumberFloatToStringLocale(
+          Utils.parseStringLocaleToNumberFloat(toAddr.amountCurrency) / zecPrice.zecPrice,
+          8,
+        );
+      } else {
+        toAddr.amount = '';
+      }
+      //console.log('update field 2', toAddr.amount, toAddr.amountCurrency);
+    }
+
+    if (memo !== null) {
+      toAddr.memo = memo;
+    }
+
+    if (includeUAMemo !== null) {
+      toAddr.includeUAMemo = includeUAMemo;
+    }
+
+    newState.toaddr = newToAddr;
+    setSendPageState(newState);
+    //console.log(newState);
+  };
+
+  useEffect(() => {
+    if (validAddress === 0 && validAmount === 0) {
+      setFee(0);
+    } else if (validAddress !== -1 && validAmount !== -1) {
+      calculateFeeWithPropose(
+        sendPageState.toaddr.amount,
+        sendPageState.toaddr.to,
+        sendPageState.toaddr.memo,
+        sendPageState.toaddr.includeUAMemo,
+      );
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    sendPageState.toaddr.amount,
+    sendPageState.toaddr.includeUAMemo,
+    //sendPageState.toaddr.memo,
+    sendPageState.toaddr.to,
+    validAddress,
+    validAmount,
+  ]);
+
   useEffect(() => {
-    const defaultFee = info.defaultFee || Utils.getFallbackDefaultFee();
     // transparent is not spendable.
     const spend = totalBalance.spendablePrivate + totalBalance.spendableOrchard;
-    const max = spendable - defaultFee;
-    if (max > 0) {
+    const max = spend - fee - (donation ? Utils.parseStringLocaleToNumberFloat(Utils.getDefaultDonationAmount()) : 0);
+    if (max >= 0) {
+      // if max is 0 then the user can send a memo with amount 0.
       setMaxAmount(max);
+      setNegativeMaxAount(false);
+      //if (sendAllClick) {
+      //  updateToField(null, Utils.parseNumberFloatToStringLocale(max, 8), null, null, null);
+      //}
+    } else {
+      // if max is less than 0 then the user CANNOT send anything.
+      setMaxAmount(0);
+      setNegativeMaxAount(true);
+      //if (sendAllClick) {
+      //  updateToField(null, '0', null, null, null);
+      //}
     }
-    setFee(defaultFee);
     setSpendable(spend);
+    //setSendAllClick(false);
 
     const stillConf =
       totalBalance.orchardBal !== totalBalance.spendableOrchard ||
       totalBalance.privateBal !== totalBalance.spendablePrivate ||
       someUnconfirmed;
     const showShield =
-      totalBalance &&
       (someUnconfirmed ? 0 : totalBalance.transparentBal) > 0 &&
-      (someUnconfirmed ? 0 : totalBalance.transparentBal) + totalBalance.spendablePrivate > info.defaultFee;
+      (someUnconfirmed ? 0 : totalBalance.transparentBal) + totalBalance.spendablePrivate > fee;
     const showUpgrade =
```

Comment:
upgrade is like sapling -> orchard ?

Reply:
shielding - from transparent to orchard.
upgrading - from sapling to orchard.